exporter_test.go

```diff
@@ -1255,7 +1255,29 @@ func TestPasswordProtectedInstance(t *testing.T) {
 		t.Skipf("TEST_PWD_REDIS_URI not set - skipping")
 	}
 	uri := os.Getenv("TEST_PWD_REDIS_URI")
-	setupDBKeys(t, uri)
+
+	e, _ := NewRedisExporter(uri, Options{Namespace: "test", Registry: prometheus.NewRegistry()})
+	ts := httptest.NewServer(e)
+	defer ts.Close()
+
+	chM := make(chan prometheus.Metric, 10000)
+	go func() {
+		e.Collect(chM)
+		close(chM)
+	}()
+
+	body := downloadURL(t, ts.URL+"/metrics")
+
+	if !strings.Contains(body, "test_up") {
+		t.Errorf("error, missing test_up")
+	}
+}
+
+func TestUserPasswordProtectedInstance(t *testing.T) {
```

Comment:
Instead of a completely new test you can also do this in a table-driven way and have the two URIs in the table, up to you, I'm fine either way.